### PR TITLE
art-notify: Include job fail rate

### PR DESCRIPTION
**Pull request by thegreyd**
Test run: https://redhat-internal.slack.com/archives/GUSEXGPJP/p1724780696787649?thread_ts=1724780696.235409&cid=GUSEXGPJP

### CODE REVIEW

File: scheduled-jobs/scanning/art-notify/art-notify.py [hunk 1]
```diff
@@ -44,25 +44,28 @@ def get_failed_jobs_text():
...
-                    # would be monitored by whoever triggered them
-                    # Builds which are triggered by another build will not have userId
-                    # so we include them by default
-                    user_id = next(a['causes'][0].get('userId') for a in build.get('actions') if a.get('causes'))
-                    if user_id and user_id != ART_BOT_JENKINS_USERID:
-                        continue
                     failed_job_ids.append(build['number'])
             if len(failed_job_ids) > 0:
-                failed_jobs.append((unquote(job_name), failed_job_ids, job['url']))
+                fail_rate = (len(failed_job_ids)/total_eligible_builds)*100
```
Comment (joepvd) [inline scheduled-jobs/scanning/art-notify/art-notify.py:67]: Always afraid for `ZeroDivisionError`. Think it is not logically possible, but would test for divisor != 0 or catch

**Comment by joepvd** [comment]
/lgtm
/approve
yet:
/hold
to optionally address comment

File: scheduled-jobs/scanning/art-notify/art-notify.py (hunk 1 above)
Comment (locriandev) [reply scheduled-jobs/scanning/art-notify/art-notify.py:67]: Looks like if `failed_job_ids > 0` we should always get `total_eligible_builds > 0`, so this should not happen

File: scheduled-jobs/scanning/art-notify/art-notify.py [hunk 2]
```diff
@@ -76,14 +79,15 @@ def slack_link(job_name, job_url, job_id=None, text=None):
...
+        failed_jobs.sort(key=lambda x: x[2], reverse=True)
         failed_jobs_list = []
-        for job_name, failed_job_ids, job_url in failed_jobs:
+        for job_name, failed_job_ids, fail_rate, job_url in failed_jobs:
             link = slack_link(job_name, job_url)
             text = f"* {link}: {len(failed_job_ids)} "
             failed_job_ids.sort(reverse=True)
             for i in range(min(3, len(failed_job_ids))):
                 text += f"[{slack_link(job_name, job_url, job_id=failed_job_ids[i], text=i+1)}] "
+            failed_text += f"Fail rate: {fail_rate:.1f}%"
```
Comment (locriandev) [inline scheduled-jobs/scanning/art-notify/art-notify.py:90]: `UnboundLocalError: local variable 'failed_text' referenced before assignment`
seen [here](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/hack/job/dpaolell/job/art-attention/15/console)

**Review by locriandev — CHANGES_REQUESTED** [review]

File: scheduled-jobs/scanning/art-notify/art-notify.py [hunk 3]
```diff
@@ -76,14 +79,15 @@ def slack_link(job_name, job_url, job_id=None, text=None):
         return f"<{link}|{text}>"
 
     if failed_jobs:
-        failed_jobs.sort(key=lambda x: len(x[1]), reverse=True)
+        failed_jobs.sort(key=lambda x: x[2], reverse=True)
```
Comment (locriandev) [inline scheduled-jobs/scanning/art-notify/art-notify.py:82]:
```suggestion
        failed_text = ''
        failed_jobs.sort(key=lambda x: x[2], reverse=True)
```

File: scheduled-jobs/scanning/art-notify/art-notify.py (hunk 2 above)
Comment (thegreyd) [inline scheduled-jobs/scanning/art-notify/art-notify.py:90]:
```suggestion
            text += f"Fail rate: {fail_rate:.1f}%"
```

File: scheduled-jobs/scanning/art-notify/art-notify.py (hunk 3 above)
Comment (thegreyd) [reply scheduled-jobs/scanning/art-notify/art-notify.py:82]: Ah I meant to add to text not failed_text

**Pull request merged**